src/EditorFeatures/Core.Wpf/QuickInfo/ContentControlService.cs

```diff
@@ -68,13 +69,13 @@ public DisposableToolTip CreateDisposableToolTip(Document document, ITextBuffer
             // 
             // our underlying preview tagger and mechanism to attach tagger to associated buffer of
             // opened document will light up automatically
-            var workspace = new PreviewWorkspace(document.Project.Solution);
-            workspace.OpenDocument(document.Id, textBuffer.AsTextContainer());
+            using var workspace = PreviewWorkspace.Create(document.Project.Solution);
+            workspace.Target.OpenDocument(document.Id, textBuffer.AsTextContainer());
 
-            return new DisposableToolTip(toolTip, workspace);
+            return DisposableToolTip.Create(toolTip, workspace);
```

Comment:
visually, this just looks weird.  It looks like you're returning something that is holding onto something that is explicitly being disposed.  needs comments to explain why this is ok.

Reply:
This is _the_ correct pattern for correctly transferring ownership of a `ReferenceCountedDisposable<T>`. An RCD passed by value is only assured to be alive until the end of the call. The callee must obtain its own reference if it needs to store the value to a field. The behavior of the caller depends on the source of the RCD: if the caller created the RCD (as is the case here) it must release it before returning. If the caller received the RCD via an argument, it doesn't own that instance and can just return.

Reply:
> This is the correct pattern for correctly transferring ownership of a ReferenceCountedDisposable<T>

literally nothing in the code states taht it's an RCD though.  It just looks like a bog standard captured disposable.  RCD is fine here because the using is only affecting the ref-count. And, when that is clear, is totally ok.  But in cases like this, there's no indication of ref-counting at all, so it immediatley jumps out as looking problematic.

Reply:
➡️ Renamed `Create` methods that return RCD to `CreateReferenceCounted`